src/metabase/models/permissions/parse.clj

```diff
@@ -76,22 +78,29 @@
     [:permission t]                (path1 t)
     [:schema-name schema-name]     (unescape-path-component schema-name)
     [:all]                         [:all] ; admin permissions
-    [:db db-id]                    (let [db-id (Long/parseUnsignedLong db-id)]
-                                     [[:db db-id :data :native :write]
-                                      [:db db-id :data :schemas :all]])
-    [:db db-id db-node]            (let [db-id (Long/parseUnsignedLong db-id)]
-                                     (into [:db db-id] (path1 db-node)))
-    [:schemas]                     [:data :schemas :all]
-    [:schemas schema]              (into [:data :schemas] (path1 schema))
+
+    [:db db-id]                    (let [db-id (Long/parseUnsignedLong db-id)] [[:db db-id :data :native :write] [:db db-id :data :schemas :all]])
+    [:db db-id db-node]            (into [:db (Long/parseUnsignedLong db-id) :data] (path1 db-node))
+
+    [:data-v2 db-id]              (let [db-id (Long/parseUnsignedLong db-id)] [[:db db-id :data :native :write]])
+    [:data-v2 db-id db-node]      (into [:db (Long/parseUnsignedLong db-id) :data] (path1 db-node))
+
+    [:query-v2 db-id]              (let [db-id (Long/parseUnsignedLong db-id)] [[:db db-id :query :native :write] [:db db-id :query :schemas :all]])
+    [:query-v2 db-id db-node]      (into [:db (Long/parseUnsignedLong db-id) :query] (path1 db-node))
```

Comment:
These now add their `:data` or `:query` key here, instead of e.g. on the r.h.s. of `[:native]`.